### PR TITLE
Test the Android embedding v2

### LINKED ISSUE #33107: Create add-to-app embedding integration test
There are already some in https://github.com/flutter/flutter/issues/24117

Ideally move to engine PRs

### PR DESCRIPTION
**Pull request by blasten**
## Description

Enables tests of the Android embedding v2.

I also renamed the flag as `--android-embedding-v2`.

## Related Issues

* Fixes https://github.com/flutter/flutter/issues/33107.
* Fixes last item on https://github.com/flutter/flutter/issues/33884

### CODE REVIEW

File: dev/bots/test.dart [hunk 1]
```diff
@@ -957,24 +960,37 @@ Future<void> _androidGradleTests(String subShard) async {
     print('No Android SDK detected or on Windows, skipping Android gradle test.');
     return;
   }
-
-  final Map<String, String> env = <String, String> {
+  final Map<String, String> defaultEnv = <String, String>{
```
Comment (jonahwilliams) [inline dev/bots/test.dart]: I would start with the assumption that this is running in a separate shard.

**Review by jonahwilliams — COMMENTED** [review]
Rename seems fine to me. Do we have a test that is running that executes flutter create and attempts to run the resulting app with the new embedding?

**Comment by blasten** [comment]
@jonahwilliams the exact combination of `flutter create` + `flutter run` doesn't exist yet.  We do run an app using the new embedding on a real device, but the code is checked in.

Most of tests added in this PR are testing the build. Once https://github.com/flutter/flutter/pull/42684 is checked in, then we can migrate/clone the example apps to test hot reload.

**Comment by jonahwilliams** [comment]
I'd be concerned that the changes to create are not tested enough. I think there might be a devicelab test that does something like that

File: dev/bots/test.dart (hunk 1 above)
Comment (blasten) [reply dev/bots/test.dart]: Done

**Comment by blasten** [comment]
discussed offline. This first PR focuses on migrated the presubmit tests.

**Review by jonahwilliams — APPROVED** [review]
Approach looks LGTM

**Pull request merged**